### DISCUSSION TITLE
Standalone compilation of HeishaMon

### DISCUSSION THREAD

**Discussion by krzbor** [Egyras/HeishaMon#576] [Q&A]
While compiling in Arduino IDE, I get the error:
Compilation error: 'MEMPOOL_ADDRESS' was not declared in this scope
It refers to line 76:

unsigned char *mempool = (unsigned char *)MEMPOOL_ADDRESS;

in the rules.cpp module.
What could be the cause of the error?

**Comment by geduxas** [comment]
Look GitHub action's, there is complete flags for compiler.

**Reply by geduxas** [reply]
arduino-cli compile --output-dir . --fqbn=esp8266:esp8266:d1_mini:xtal=160,vt=flash,ssl=basic,mmu=3216,non32xfer=fast,eesz=4M2M,ip=lm2f,dbg=Disabled,lvl=None____,wipe=none,baud=921600 --warnings=none --verbose HeishaMon.ino

**Comment by krzbor** [comment]
I am receiving the error:
Error during build: Invalid FQBN: getting build properties for board esp8266:esp8266:d1_mini: invalid option 'non32xfer'

**Comment by krzbor** [comment]
I removed two parameters: "non32xfer=fast" and "mmu=3216"
However, an error appeared:

c:\prog\arduino\HeishaMon\rules.cpp:76:43: error: 'MEMPOOL_ADDRESS' was not declared in this scope
unsigned char *mempool = (unsigned char *)MEMPOOL_ADDRESS;

Used libraries:
    ESP8266WiFi: 1.0
    ESP8266mDNS: 1.2
    DNSServer: 1.1.1
    ArduinoOTA: 1.0
    ArduinoJson: 7.2.0
    PubSubClient: 2.8
    LittleFS (esp8266): 0.1.0
    OneWire: 2.3.8
    DallasTemperature: 3.9.0

Used platform:
    esp8266: 2.7.4

Has anyone managed to compile the code shared on GitHub?

**Reply by geduxas** [reply]
yes, its compiles without any problem in arduino IDE and via command line, also on windows and linux. No problems for me.

**Comment by krzbor** [comment]
So why is it returning this error? Where is 'MEMPOOL_ADDRESS' defined? Could you install the Arduino IDE in a Windows sandbox and, after downloading the libraries, try compiling it? If it works for you, that would mean there's something wrong on my end.

**Reply by geduxas** [reply]
Maybe same issue as in #524 ?

**Comment by krzbor** [comment]
The problem is that I had to remove MMU from the command line; otherwise, the compilation wouldn’t start. I also don’t have MMU in the IDE. Have you tried compiling with the latest arduino-cli?
![mmu](https://github.com/user-attachments/assets/7bc366b3-1a72-437f-ad84-7de451d634e8)

**Reply by krzbor** [reply]
I am using this version of arduino-cli:
https://downloads.arduino.cc/arduino-cli/arduino-cli_latest_Windows_64bit.zip

**Comment by krzbor** [comment]
The problem has been solved. The ESP8266 library version 2.7.4 is too old. The compilation completed successfully after installing library version 3.1.2.

**Comment by geduxas** [comment]
Mostly i am using linux for compile.. and using visual studio with platformio.. also let compile in github automations..

**Reply by krzbor** [reply]
What options should be set in the "arduino-cli upload" command in the "--fqbn" parameter? Are these the same options as for compilation?
